### PR TITLE
Improve plug'n'playability of TS declarations

### PR DESCRIPTION
**Pull request by weswigham**
This declaration file uses the esnext async iterable types - if using [TS >= 3.0, this comment will pull that lib in automatically.](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#-reference-lib--reference-directives)

Related: #1189

### CODE REVIEW

**Comment by weswigham** [comment]
Geh, older versions of TS issue an error on the directive - just keep this in mind if you update to a newer TS version.

**Comment by gr2m** [comment]
Okay so updating typescript to latest version should take care of it? Can you add it to this PR?

File: scripts/templates/index.d.ts.tpl [hunk 1]
```diff
@@ -2,6 +2,8 @@
  * This declaration file requires TypeScript 2.1 or above.
```
Comment (wolfy1339) [inline scripts/templates/index.d.ts.tpl]:
```suggestion
  * This declaration file requires TypeScript 3.0 or above.
```

**Review by wolfy1339 — CHANGES_REQUESTED** [review]

**Review by gr2m — APPROVED** [review]
Thanks everyone!

**Pull request merged**